Carbon.Domain.EntityFrameworkCore.Extensions/EFExtensions.cs

```diff
@@ -348,5 +349,58 @@ public class RelationEntityPair<T, U>
             }
             return relationEntities;
         }
+
+        public static async Task<List<T>> ToListEntityFilteredWithIncludeAsync<T, U>(
+            this IQueryable<T> relationEntities,
+            List<PermissionDetailedDto> roleDetails,
+            List<Guid> solutionFilter,
+            DbContext ctx)
+            where T : IHaveOwnership<U>, IEntity
+            where U : EntitySolutionRelation
+        {
+            return await relationEntities
+                .IncludeOwnershipFilter<T, U>(roleDetails)
+                .IncludeSolutionRelation<T, U>(ctx)
+                .ToListEntityFilteredWithSolutionAsync(solutionFilter);
+        }
+        
+        public static async Task<PagedList<T>> ToPagedListEntityFilteredWithIncludeAsync<T, U>(
+            this IQueryable<T> relationEntities,
+            BaseRequestPageDto dto,
+            List<PermissionDetailedDto> roleDetails,
+            List<Guid> solutionFilter,
+            DbContext ctx)
+            where T : IHaveOwnership<U>, IEntity
+            where U : EntitySolutionRelation
+        {
+            var entities = await relationEntities.ToListEntityFilteredWithIncludeAsync<T, U>(roleDetails, solutionFilter, ctx);
```

Comment:
This takes all the data and does pagination. Is this the intended use?

Reply:
The query is accepted as reduced to certain parameters beforehand. Here, it passes through the ownership filter, followed by paging. ToListEntityFilteredWithIncludeAsync method is called to apply the ownership filter. 

It indirectly goes to the ToListEntityFilteredWithSolutionAsync method. As you can see in this method, ToList is used to apply the ownership filter. So it already gets all the data from here. This is partly true. Do you have a different solution?

Reply:
You're inside the library, so you need to do this pagination without pulling all the data here.

Reply:
Instead, I added a method named ToPagedListEntityFilteredWithIncludeAsync. Can you check it?